### ISSUE TITLE
Moirai2 forecast accuracy drops significantly when `feat_dynamic_real` are present

### ISSUE THREAD

**Issue by shchur** [SalesforceAIResearch/uni2ts#224]
**Describe the bug**
I am not sure if this should be considered a bug or a general question, but I have observed that the forecast accuracy of Moirai2 consistently and noticeably drops when `feat_dynamic_real` are provided.

I am not sure if this is a problem with, e.g., how I provide the data to the model using the `PandasDataset`, or if this is an issue with the model itself.

**To Reproduce**

Here is a representative example using the Electricity Price Forecasting - FR dataset, but I have observed similar behavior on at least 10 other datasets involving covariates.

```python
# !pip install datasets matplotlib "uni2ts @ git+https://github.com/SalesforceAIResearch/uni2ts@b472ef6dcf7"
import datasets
import matplotlib.pyplot as plt
from gluonts.dataset.pandas import PandasDataset
from uni2ts.model.moirai2 import Moirai2Forecast, Moirai2Module

h = 48
context_length = 1024
ds = datasets.load_dataset("autogluon/fev_datasets", "epf_electricity_fr", split="train")
df = ds.to_pandas().explode(["timestamp", "target", "Generation forecast", "System load forecast"])
df = df.query("timestamp < '2016-11-29'").tail(context_length + h)
gluonts_dataset = PandasDataset.from_long_dataframe(
    df,
    item_id="id",
    timestamp="timestamp",
    target="target",
    feat_dynamic_real=["Generation forecast", "System load forecast"],
    future_length=h,
)

past = df.iloc[:-h]
future = df.iloc[-h:]

plt.figure(figsize=[20, 5])
plt.plot(past["timestamp"], past["target"], label="y_past", c="C0")
plt.plot(future["timestamp"], future["target"], label="y_true", c="C0")
for feat_dynamic_real_dim in [0, 2]:
    model = Moirai2Forecast(
        module=Moirai2Module.from_pretrained("Salesforce/moirai-2.0-R-small"),
        prediction_length=h,
        context_length=context_length,
        target_dim=1,
        feat_dynamic_real_dim=feat_dynamic_real_dim,
        past_feat_dynamic_real_dim=0,
    )
    predictor = model.create_predictor(batch_size=1, device="cpu")
    forecast = list(predictor.predict(gluonts_dataset))[0]
    plt.plot(future["timestamp"], forecast.quantile(0.5), label=f"Forecast w/ {feat_dynamic_real_dim=}", c=f"C{feat_dynamic_real_dim+1}")
plt.grid()
plt.ylim(0, 200)
plt.legend()
plt.show()
```

Output
```
feat_dynamic_real_dim=0, MAE=5.098
feat_dynamic_real_dim=2, MAE=16.330
```
<img width="1640" height="434" alt="Image" src="https://github.com/user-attachments/assets/932cf3fa-a514-4a0c-8251-43d806564f36" />

**Expected behavior**

Forecast accuracy does not degrade when covariates are used.

**Environment**
* Operating system: Linux
* Python version: 3.11
* PyTorch version: 2.4.1+cu121
* uni2ts version (have you pulled the latest version?): Install from source, commit b472ef6dcf7

**Comment by sharky26** [comment]
me too on electricity price PJM dataset

**Comment by chenghaoliu89** [comment]
Hi @shchur the current version of moirai2  did not include any covariate pre-training data.

**Comment by sharky26** [comment]
> Hi [@shchur](https://github.com/shchur) the current version of moirai2 did not include any covariate pre-training data.

Hi, thanks for reply. Besides, I wonder if in a multivariable forecasting scenario, the multiple input variables will be treated seperately as several univariate time series or there will be interaction among them?

**Comment by chenghaoliu89** [comment]
Hi @sharky26, current moirai2 is pre-trained with univariate data, and it does not learn any cross-variate patterns. You can check the notebooks for the inference with multivaraite data, which support both ways (treated seperately as several univariate or as a multivaraite ts).

**Comment by Claire-bx** [comment]
> > Hi [@sharky26](https://github.com/sharky26), current moirai2 is pre-trained with univariate data, and it does not learn any cross-variate patterns. You can check the notebooks for the inference with multivaraite data, which support both ways (treated seperately as several univariate or as a multivaraite ts).
> 
> thank you for response 😊 I‘m looking for Large Time Series Model method that can handle covariates within the model, rather than treating them as exogenous variables like chronos and timsFM. This work is of great value!

Hi, I have a question regarding the model input specification.

For Moirai, should the features in feat_dynamic_real also be included in past_feat_dynamic_real?

Currently, I set:
```
past_feat_dynamic_real = conditional_columns + control_columns + observable_columns
feat_dynamic_real = control_columns + observable_columns
```

**Comment by sharky26** [comment]
> > > Hi [@sharky26](https://github.com/sharky26), current moirai2 is pre-trained with univariate data, and it does not learn any cross-variate patterns. You can check the notebooks for the inference with multivaraite data, which support both ways (treated seperately as several univariate or as a multivaraite ts).
> > 
> > 
> > thank you for response 😊 I‘m looking for Large Time Series Model method that can handle covariates within the model, rather than treating them as exogenous variables like chronos and timsFM. This work is of great value!
> 
> Hi, I have a question regarding the model input specification.
> 
> For Moirai, should the features in feat_dynamic_real also be included in past_feat_dynamic_real?
> 
> Currently, I set:
> 
> ```
> past_feat_dynamic_real = conditional_columns + control_columns + observable_columns
> feat_dynamic_real = control_columns + observable_columns
> ```

@Claire-bx Hi, I think there is no need to include feat_dynamic_real in past_feat_dynamic_real repeatedly. You may refer to [https://github.com/SalesforceAIResearch/uni2ts/blob/main/example/moirai_forecast_pandas.ipynb](url)
